app/controllers/users_controller.rb

```
@@ -0,0 +1,32 @@
+class UsersController < ApplicationController
+  before_action :find_user_or_redirect, only: [:show]
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      redirect_to @user, flash: { success: 'Welcome to the Sample App!' }
+    else
+      render 'new', status: :unprocessable_entity
+    end
+  end
+
+  def show; end
+
+  private
+
+  def user_params
+    params.require(:user).permit(:name, :email, :password, :password_confirmation)
+  end
+
+  def find_user_or_redirect
+    if (@user = User.find_by(id: params[:id]))
+      @user
+    else
+      flash[:danger] = 'The user profile you are trying to access could not be found.'
```

Awesome! Good work. Sometimes its okay to simply let it fail as RecordNotFound will raise and result in translation to a 404. But if you want to provide the user a slightly better experience this is a solid approach.